packages/driver/src/cypress/cy.ts

```diff
@@ -549,24 +547,44 @@ export class $Cy extends EventEmitter2 implements ITimeouts, IStability, IAssert
       try {
         const autWindow = getContentWindow($autIframe)
 
-        setWindowDocumentProps(autWindow, this.state)
+        let isSameOrigin
 
-        // we may need to update the url now
-        this.urlNavigationEvent('load')
+        try {
+          autWindow.location.origin
+          isSameOrigin = true
+        } catch (err: any) {
+          if (!this.config('experimentalSessionAndOrigin') && err.name === 'SecurityError') {
+            throw err
+          }
 
-        // we normally DONT need to reapply contentWindow listeners
-        // because they would have been automatically applied during
-        // onBeforeAppWindowLoad, but in the case where we visited
-        // about:blank in a visit, we do need these
-        this.contentWindowListeners(autWindow)
+          isSameOrigin = false
+        }
+
+        if (isSameOrigin) {
+          setWindowDocumentProps(autWindow, this.state)
+
+          // we may need to update the url now
+          this.urlNavigationEvent('load')
+
+          // we normally DON'T need to reapply contentWindow listeners
+          // because they would have been automatically applied during
+          // onBeforeAppWindowLoad, but in the case where we visited
+          // about:blank in a visit, we do need these
+          this.contentWindowListeners(autWindow)
+        } else {
+          this.state('window', autWindow)
+          // we may need to update the url now
+          this.urlNavigationEvent('load')
+        }
 
         // stability is signalled after the window:load event to give event
         // listeners time to be invoked prior to moving on, but not if
         // there is a cross-origin error and the cy.origin API is
         // not utilized
         try {
-          this.Cypress.action('app:window:load', this.state('window'))
-          const remoteLocation = this.getRemoteLocation()
+          const remoteLocation = await this.getCrossOriginRemoteLocation()
```

Comment:
Can we do this?

```suggestion
          const remoteLocationHref = await this.getCrossOriginRemoteLocation('href')
```

Reply:
we could, but we use the origin policy too.